query-engine/connectors/sql-query-connector/src/query_builder/read.rs

```diff
@@ -29,8 +29,8 @@ impl SelectDefinition for Select<'static> {
 
 impl SelectDefinition for QueryArguments {
     fn into_select(self, model: &ModelRef) -> Select<'static> {
-        let (table_opt, cursor_condition) = cursor_condition::build(&self, &model);
-        let (orderings, joins) = ordering::build(&self, &model);
+        let (orderings, ordering_joins) = ordering::build(&self, &model);
+        let (table_opt, cursor_condition) = cursor_condition::build(&self, &model, &ordering_joins);
```

Comment:
Joins computed by `ordering::build` are now passed to the `cursor_condition::build` method